pkg/install/start_vms.go

```diff
@@ -0,0 +1,74 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// startVMs checks cluster VMs power state and starts deallocated and stopped VMs, if any
+func (i *Installer) startVMs(ctx context.Context) error {
+	resourceGroupName := stringutils.LastTokenByte(i.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	vms, err := i.virtualmachines.List(ctx, resourceGroupName)
+	if err != nil {
+		return err
+	}
+
+	{
+		g, groupCtx := errgroup.WithContext(ctx)
+		for idx, vm := range vms {
```

Comment:
as discussed, let's go with this for now.  I still worry about the case of lots of parallel admin upgrades * lots of VMs * getting 409s from Azure.